daffodil-runtime1/src/main/scala/edu/illinois/ncsa/daffodil/processors/parsers/PState.scala

```diff
@@ -284,8 +284,53 @@ final class PState private (
     dataInputStream.setDebugging(flag)
   }
 
+  /**
+   * Checks for bit order change. If the bit order is changing, checks if we're
+   * on a proper byte boundary.
+   */
   final override protected def checkBitOrder(): Unit = {
-    ParserBitOrderChecks.checkParseBitOrder(this)
```

Comment:
All I did here is get rid of the objects ParserBitOrderChecks and UnparserBitOrderChecks and made those methods of PState and UState respectively.